From: Hannes Reinecke <hare@suse.de>
Subject: Do not rescan for partitions on CF cards with PCMCIA adapters

When using removable drives on ide, the ide layer will always trigger an
partition rescan as it assumes that any disk change cannot reliably reported.
This results in an endless hotplug event loop as we need to access the drive
in order to report in properly to userland.

However with PCMCIA CF adapters it's very unlikely that the card is ejected without
the adapter; and removing the adapter will destroy all references to that disk, including
the IDE stack.
And as the only removable ide drivers which are identified as flash cards are indeed CF cards 
on a PCMCIA adapter, we can (and indeed should) no rescan the partitions.


--- linux-2.6.11/drivers/ide/ide-disk.c.orig	2005-03-17 10:19:08.000000000 +0100
+++ linux-2.6.11/drivers/ide/ide-disk.c	2005-03-17 11:03:33.000000000 +0100
@@ -1289,7 +1289,11 @@
 		drive->attach = 0;
 		return 0;
 	}
+	/* We get reliably notified from PCMCIA layer */
+	if (drive->is_flash)
+		return 0;
+
 	/* if removable, always assume it was changed */
 	return drive->removable;
 }
